lib/include/openamp/rpmsg_virtio.h

```diff
@@ -141,8 +141,8 @@ rpmsg_virtio_create_virtqueues(struct rpmsg_virtio_device *rvdev,
 			       const char *names[],
 			       vq_callback *callbacks)
 {
-	return virtio_create_virtqueues(rvdev->vdev, flags, nvqs, names,
-					callbacks);
+	return rvdev->vdev->func->create_virtqueues(rvdev->vdev, flags, nvqs,
+						    names, callbacks);
```

Comment:
In fact doing that you break some project compatibility:
for instance
https://elixir.bootlin.com/zephyr/latest/source/subsys/ipc/rpmsg_service/rpmsg_backend.c#L145

At least need to support legacy using deprecating method.

Reply:
let's check create_virtqueues and do the different dispatch

Reply:
yes something like 
```
if (rvdev->vdev->func->create_virtqueues) {
	return rvdev->vdev->func->create_virtqueues(rvdev->vdev, flags, nvqs,
						    names, callbacks);
} else {
	metal_log(METAL_LOG_WARNING,
				  "deprecated method please set create_virtqueues instead");
	virtio_create_virtqueues(rvdev->vdev, flags, nvqs, names,
					callbacks);
}
```

Reply:
This has been implmented in other PR: https://github.com/OpenAMP/open-amp/pull/495